Flaky test: test_stragglers

From a [recent run](https://github.com/cubed-dev/cubed/actions/runs/10308706981/job/28536745015?pr=548):


```
______________________ test_stragglers[timing_map0-10-2] _______________________

tmp_path = PosixPath('/tmp/pytest-of-runner/pytest-0/test_stragglers_timing_map0_100')
timing_map = {0: [10]}, n_tasks = 10, retries = 2

    @pytest.mark.parametrize(
        "timing_map, n_tasks, retries",
        [
            ({0: [10]}, 10, 2),
        ],
    )
    # fmt: on
    def test_stragglers(tmp_path, timing_map, n_tasks, retries):
        outputs = asyncio.run(
            run_test(
                function=partial(deterministic_failure, tmp_path, timing_map),
                input=range(n_tasks),
                retries=retries,
                use_backups=True,
            )
        )
    
        assert outputs == set(range(n_tasks))
    
>       check_invocation_counts(tmp_path, timing_map, n_tasks, retries)

cubed/tests/runtime/test_local.py:102: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

path = PosixPath('/tmp/pytest-of-runner/pytest-0/test_stragglers_timing_map0_100')
timing_map = {0: [10]}, n_tasks = 10, retries = 2
expected_invocation_counts_overrides = None

    def check_invocation_counts(
        path, timing_map, n_tasks, retries=None, expected_invocation_counts_overrides=None
    ):
        expected_invocation_counts = {}
        for i in range(n_tasks):
            if i not in timing_map:
                expected_invocation_counts[i] = 1
            else:
                timing_codes = timing_map[i]
                expected_invocation_count = len(timing_codes) + 1
    
                if retries is not None:
                    # there shouldn't have been more than retries + 1 invocations
                    max_invocations = retries + 1
                    expected_invocation_count = min(
                        expected_invocation_count, max_invocations
                    )
    
                expected_invocation_counts[i] = expected_invocation_count
    
        if expected_invocation_counts_overrides is not None:
            expected_invocation_counts.update(expected_invocation_counts_overrides)
    
        # retrieve outputs concurrently, so we can test on large numbers of inputs
        # see https://filesystem-spec.readthedocs.io/en/latest/async.html#synchronous-api
        if re.match(r"^[a-zA-Z]:\\", str(path)):  # Windows local file
            protocol = ""
        else:
            protocol = urlparse(str(path)).scheme
        fs = fsspec.filesystem(protocol)
        paths = [join_path(path, str(i)) for i in range(n_tasks)]
        out = fs.cat(paths)
        path_to_i = lambda p: int(p.rsplit("/", 1)[-1])
        actual_invocation_counts = {path_to_i(path): int(val) for path, val in out.items()}
    
        if actual_invocation_counts != expected_invocation_counts:
            for i, expected_count in expected_invocation_counts.items():
                actual_count = actual_invocation_counts[i]
                if actual_count != expected_count:
                    print(
                        f"Invocation count for {i}, expected: {expected_count}, actual: {actual_count}"
                    )
>       assert actual_invocation_counts == expected_invocation_counts
E       AssertionError

cubed/tests/runtime/utils.py:104: AssertionError
=========================== short test summary info ============================
FAILED cubed/tests/runtime/test_local.py::test_stragglers[timing_map0-10-2] - AssertionError
```

--- a/cubed/tests/runtime/utils.py
+++ b/cubed/tests/runtime/utils.py
@@ -17,7 +17,7 @@ def write_int_to_file(path, i):
         f.write(str(i))
 
 
-def deterministic_failure(path, timing_map, i, *, name=None):
+def deterministic_failure(path, timing_map, i, *, default_sleep=0.01, name=None):
     """A function that can either run normally, run slowly, or raise
     an exception, depending on input and invocation count.
 
@@ -27,7 +27,8 @@ def deterministic_failure(path, timing_map, i, *, name=None):
     the sign indicates the input is returned normally (positive, or 0),
     or an exception is raised (negative).
 
-    If a input is missing then all invocations will run normally.
+    If a input is missing then all invocations will run normally, with a
+    small default sleep to avoid spurious backups being launched.
 
     If there are subsequent invocations to the ones in the sequence, then
     they will all run normally.
@@ -41,11 +42,11 @@ def deterministic_failure(path, timing_map, i, *, name=None):
         invocation_count = 0
     write_int_to_file(invocation_count_file, invocation_count + 1)
 
-    timing_code = 0
+    timing_code = default_sleep
     if i in timing_map:
         timing_codes = timing_map[i]
         if invocation_count >= len(timing_codes):
-            timing_code = 0
+            timing_code = default_sleep
         else:
             timing_code = timing_codes[invocation_count]
 

